boards: doc: Include only index documents to toctree

finikorg opened the pull request:
Use correct glob pattern to exclude common rst helpers from toctree.

Fixes current issues with RST helpers added to TOC here: https://docs.zephyrproject.org/latest/boards/x86/index.html

finikorg committed:
boards: doc: Include only index document to toctree

Use correct glob pattern to exclude common rst helpers from toctree.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jhedberg commented:
@finikorg there seems to be a CI failure in the documentation build:
```
[1/2] Running Sphinx HTML build...
/runner/_work/zephyr/zephyr/doc/_build/src/boards/x86/common/efi_boot.rst: WARNING: document isn't included in any toctree
/runner/_work/zephyr/zephyr/doc/_build/src/boards/x86/common/net_boot.rst: WARNING: document isn't included in any toctree
FAILED: CMakeFiles/html 
```

finikorg committed:
boards: x86: doc: Orphan included RST helpers

Mark documents, which are included in other documents as :orphan:
following existing samples in Zephyr tree.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented:
> @finikorg there seems to be a CI failure in the documentation build:
> 
> ```
> [1/2] Running Sphinx HTML build...
> /runner/_work/zephyr/zephyr/doc/_build/src/boards/x86/common/efi_boot.rst: WARNING: document isn't included in any toctree
> /runner/_work/zephyr/zephyr/doc/_build/src/boards/x86/common/net_boot.rst: WARNING: document isn't included in any toctree
> FAILED: CMakeFiles/html 
> ```

Yes, noticed, added fix for that.

kartben reviewed (approved):
sorry that I didn't review earlier.
